packages/client/src/api/client.ts

```diff
@@ -21,1907 +18,72 @@ export interface XataApiClientOptions {
-    region: string;
-    database: Schemas.DBName;
-    branch: Schemas.BranchName;
-    table: Schemas.TableName;
-    query: string;
-    fuzziness?: Schemas.FuzzinessExpression;
-    target?: Schemas.TargetExpression;
-    prefix?: Schemas.PrefixExpression;
-    filter?: Schemas.FilterExpression;
-    highlight?: Schemas.HighlightExpression;
-    boosters?: Schemas.BoosterExpression[];
-  }): Promise<Responses.SearchResponse> {
-    return operationsByTag.searchAndFilter.searchTable({
-      pathParams: { workspace, region, dbBranchName: `${database}:${branch}`, tableName: table },
-      body: { query, fuzziness, target, prefix, filter, highlight, boosters },
-      ...this.extraProps
-    });
-  }
-
-  public searchBranch({
-    workspace,
-    region,
-    database,
-    branch,
-    tables,
-    query,
-    fuzziness,
-    prefix,
-    highlight
-  }: {
-    workspace: Schemas.WorkspaceID;
-    region: string;
-    database: Schemas.DBName;
-    branch: Schemas.BranchName;
-    tables?: (
-      | string
-      | {
-          table: string;
-          filter?: Schemas.FilterExpression;
-          target?: Schemas.TargetExpression;
-          boosters?: Schemas.BoosterExpression[];
+type UserProps = {
+  headers?: Record<string, unknown>;
+};
+
+type XataApiProxy = {
+  [Tag in keyof typeof operationsByTag]: {
+    [Method in keyof (typeof operationsByTag)[Tag]]: (typeof operationsByTag)[Tag][Method] extends infer Operation extends (
+      ...args: any
+    ) => any
+      ? Omit<Parameters<Operation>[0], keyof ApiExtraProps> extends infer Params
+        ? RequiredKeys<Params> extends never
+          ? (params?: FlattenObject<Params> & UserProps) => ReturnType<Operation>
+          : (params: FlattenObject<Params> & UserProps) => ReturnType<Operation>
+        : never
+      : never;
+  };
+};
+
+const buildApiClient = () =>
+  class {
+    constructor(options: XataApiClientOptions = {}) {
+      const provider = options.host ?? 'production';
+      const apiKey = options.apiKey ?? getAPIKey();
+      const trace = options.trace ?? defaultTrace;
+      const clientID = generateUUID();
+
+      if (!apiKey) {
+        throw new Error('Could not resolve a valid apiKey');
+      }
+
+      const extraProps: ApiExtraProps = {
+        apiUrl: getHostUrl(provider, 'main'),
+        workspacesApiUrl: getHostUrl(provider, 'workspaces'),
+        fetch: getFetchImplementation(options.fetch),
+        apiKey,
+        trace,
+        clientName: options.clientName,
+        xataAgentExtra: options.xataAgentExtra,
+        clientID
+      };
+
+      return new Proxy(this, {
+        get: (_target, namespace: keyof typeof operationsByTag) => {
+          if (operationsByTag[namespace] === undefined) {
+            return undefined;
+          }
+
+          return new Proxy(
+            {},
+            {
+              get: (_target, operation: keyof (typeof operationsByTag)[keyof typeof operationsByTag]) => {
```

Comment:
Does this intercept  all operations/methods inside each API class? I guess the fetch functions that these operations/methods call are going to be intercepted?

Reply:
The `get` here does not refer to HTTP `GET` operation. The `get` in a `Proxy` is trying to access a property/field in an object like `foo.bar`. In this particular case, in the class it's intercepting trying to read any member of it.